### PR TITLE
Use Python 3 from Homebrew for macOS Travis builds

### PR DESCRIPTION
**Pull request by Kwpolska**
cc @gwax, @felixfontein — I think this will be better, but feel free to disagree

I’ll merge this if nobody complains by Friday morning (UTC).

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -20,13 +20,14 @@ matrix:
   include:
   - os: osx
     language: generic
-    env: NMODE=nikola
+    env: NMODE=nikola OS=macOS
```
Comment (gwax) [inline .travis.yml]: The apple vs penguin icons aren't enough?

File: .travis.yml [hunk 2]
```diff
@@ -20,13 +20,14 @@ matrix:
...
     before_install:
-    - brew upgrade pyenv
-    - eval "$(pyenv init -)"
-    - pyenv install 3.6.1
-    - pyenv global 3.6.1
+    - brew install python3
+    - pip3 install virtualenv
+    - python3 -m virtualenv nikola-tests-venv
+    - source nikola-tests-venv/bin/activate
+    - pip install --upgrade pip
```
Comment (Kwpolska) [inline .travis.yml]: Good point, fixed.

File: .travis.yml (hunk 1 above)
Comment (Kwpolska) [inline .travis.yml]: They’re hard to spot, especially if you’ve never looked there before (and the icons are barely-visible grey)

Comment (gwax) [inline .travis.yml]: I asked the question expecting (and largely agreeing with) that answer.

**Pull request merged**